File: Baekjoon/Class4/12.1991 트리 순회.js
const inputFile = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = require("fs").readFileSync(inputFile, "utf8").split("\n");

const EMPTY_NODE = ".";

const tree = new Map();
const preorderOut = [];
const inorderOut = [];
const postorderOut = [];

const getOrder = (node) => {
    if(node == EMPTY_NODE) return;
    preorderOut.push(node);
    getOrder(tree.get(node)[0]);
    inorderOut.push(node);
    getOrder(tree.get(node)[1]);
    postorderOut.push(node);
};

const n = Number(input[0]);

for(let i = 1; i <= n; i++) { // 이진 트리
    const [parent, left, right] = input[i].trim().split(" ");
    tree.set(parent, [left, right]);
}

getOrder("A"); // 탐색
console.log([preorderOut.join(""), inorderOut.join(""), postorderOut.join("")].join("\n"));

File: Baekjoon/Class4/12.1991 트리 순회.py
import sys
input = sys.stdin.readline

EMPTY_NODE = "."

tree = {}
preorder_out = []
inorder_out = []
postorder_out = []

def get_order(node):
    if node == EMPTY_NODE: return
    preorder_out.append(node)
    get_order(tree[node][0])
    inorder_out.append(node)
    get_order(tree[node][1])
    postorder_out.append(node)

def main():
    n = int(input())

    for _ in range(n): #이진 트리
        parent, left, right = input().rstrip().split()
        tree[parent] = (left, right)

    get_order("A") #탐색
    print("\n".join(("".join(preorder_out), "".join(inorder_out), "".join(postorder_out))))

main()